lib/libutils/isoc/bget_malloc.c

```diff
@@ -609,7 +613,11 @@ void *mdbg_malloc(const char *fname, int lineno, size_t size)
 {
 	struct mdbg_hdr *hdr;
 
+#if defined(ARM64)
+	COMPILE_TIME_ASSERT(sizeof(struct mdbg_hdr) == sizeof(uint32_t) * 6);
```

Comment:
No it's not performance related, the C standard say this of memory returned from `malloc()` and friends:
*The pointer returned if the allocation succeeds is suitably aligned so that it may be assigned to
a pointer to any type of object with a fundamental alignment requirement and then used to access such an object or an array of such objects in the space allocated (until the space is explicitly deallocated).*
For arm32 the required alignment is 8, I'm not sure whether it's 8 or 16 for arm64.

In any case, the `struct mdbg_hdr` is added in front of the buffers allocated internally. When adding that it must still keep a good alignment.